src/float_parser.ts

```diff
@@ -0,0 +1,39 @@
+import BN from "bn.js";
+
+const buf = new ArrayBuffer(8);
+const view = new DataView(buf);
+
+export function numberToFraction(number: number) {
+  // retrieve mantissa, exponent, and sign from bit representation of number
+  view.setFloat64(0, number);
+  const bits = view.getBigUint64(0);
+  const sign = (bits >> BigInt(63)) ? BigInt(-1) : BigInt(1);
+  const exponent = ((bits >> BigInt(52)) & BigInt(0x7ff)) - BigInt(1023);
+  const one = BigInt(1) << BigInt(52);
+  const mantissa = bits & (one - BigInt(1));
+  // number is 1.mantissa * 2**exponent
+
+  switch (exponent) {
+    case BigInt(1024): // infinities and NaN
+      throw Error("Invalid number")
+    case BigInt(-1023):
+      if (mantissa == BigInt(0)) // positive and negative zero
+        return [BigInt(0), sign]
+      else // subnormal numbers
+        throw Error("Subnormal numbers are not supported") 
+  }
+
+  const mantissa_plus_one = mantissa + one
+  const shifted_exponent = exponent - BigInt(52)
+
+  if (shifted_exponent >= BigInt(0))
+    return  [
```

Comment:
Should this method directly return a Fraction?

Reply:
I could do that, but this means adding two extra `new BN((` ... `).toString())` and one `new Fraction(` ... `)` for each of the (three) return statements, which in my opinion makes the function slightly harder to read. Since this function is now a private method of the class, I'd say the change wouldn't have many benefits and it's better to format the output outside this function.